File: update_csv.py
import csv
import os

paths = [
    'continnum_db/facturas_hoteles_1000.csv',
    'facturas_hoteles_1000.csv'
]

cols_to_remove = {'precio_desayuno', 'extras_total', 'tax_rate', 'precio_tax', 'tax_id'}

for p in paths:
    if os.path.exists(p):
        with open(p, 'r', encoding='utf-8') as f:
            reader = csv.reader(f)
            header = next(reader)
            
            indices_to_keep = [i for i, h in enumerate(header) if h.strip() not in cols_to_remove]
            new_header = [header[i] for i in indices_to_keep]
            
            rows = []
            for row in reader:
                # Ensure the row has the correct number of columns
                if len(row) > max(indices_to_keep, default=-1):
                    rows.append([row[i] for i in indices_to_keep])
                    
        with open(p, 'w', encoding='utf-8', newline='') as f:
            writer = csv.writer(f)
            writer.writerow(new_header)
            writer.writerows(rows)
        print('Updated ' + p)

File: update_csv.js
const fs = require('fs');

const paths = [
    'c:/Users/Arnau/Desktop/Agentes/anything-llm/continnum_db/facturas_hoteles_1000.csv',
    'c:/Users/Arnau/Desktop/Agentes/anything-llm/facturas_hoteles_1000.csv'
];

paths.forEach(p => {
    if (fs.existsSync(p)) {
        const content = fs.readFileSync(p, 'utf8');
        // Handle potential CRLF
        const lines = content.replace(/\r/g, '').split('\n').filter(l => l.trim() !== '');
        if (lines.length === 0) return;
        
        const headers = lines[0].split(',');
        const colsToRemove = ['precio_desayuno', 'extras_total', 'tax_rate', 'precio_tax', 'tax_id'];
        
        const indicesToKeep = headers
            .map((h, i) => colsToRemove.includes(h.trim()) ? -1 : i)
            .filter(i => i !== -1);
            
        const newLines = lines.map(line => {
            const parts = line.split(',');
            return indicesToKeep.map(i => parts[i]).join(',');
        });
        
        fs.writeFileSync(p, newLines.join('\n'));
        console.log('Updated ' + p);
    }
});
